.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "external/metal"]
```

Comment:
@trws agreed.
I have become a big fan of submodules for this reason... as long as the build system also allows pointing at a user-specified path for that library